packages/babel-parser/src/parser/expression.js

```diff
@@ -891,11 +884,6 @@ export default class ExpressionParser extends LValParser {
       );
     }
 
-    // we found an async arrow function so let's not allow any inner parens
-    if (possibleAsyncArrow && innerParenStart && this.shouldParseAsyncArrow()) {
-      this.unexpected();
```

Comment:
Instead of throwing unexpected errors on
```js
var foo = async ((foo)) => {};
```
we can now recover and raise this error in `toAssignable`, which I think is more helpful

> "SyntaxError: Invalid parenthesized assignment pattern (1:18)"